tests/testsuite/ssh.rs

```diff
@@ -419,7 +416,11 @@ fn invalid_github_key() {
         .build();
     p.cargo("fetch")
         .with_status(101)
-        .with_stderr_contains("  error: SSH host key has changed for `github.com`")
+        .with_stderr_contains(if uses_gitoxide() {
+            "  git@github.com: Permission denied (publickey)."
+        } else {
+            "  error: SSH host key has changed for `github.com`"
```

Comment:
These read like two different error causes.  Which is more correct?
- Gitoxide makes it sounds like the publickey doesn't have high enough permissions
- The git2 error is about the host key changing

Reply:
Currently `gitoxide` shells the remote connection out to `ssh` and friends which makes proper error cause retrieval difficult. For now it's implemented with a separate thread scanning stderr for clues to then let the transport on stdout fail with a particular `std::io::Error`. That way it's possible to satisfy the tests, but the error message is the one that was provided by the `ssh` binary itself. It *appears* that `ssh` reports things differently if no terminal is attached.

Once the native SSH transport implementation is available it should be no problem to match the clarity that is present now.

I don't think there is anything I can do right now, maybe I am missing something though.